File: Qb.Net/SqlOm/CaseTerm.cs
using System;

namespace Viten.QueryBuilder.SqlOm
{
  /// <summary>
  /// Encapsulates a single WHEN ... THEN ... statement
  /// </summary>
  public class CaseTerm 
  {
    public CaseTerm()
    {
    }

    /// <summary>
    /// Creates a new CaseTerm
    /// </summary>
    /// <param name="condition">Condition for the WHEN clause</param>
    /// <param name="val">Value for the THEN clause</param>
    public CaseTerm(WhereClause condition, OmExpression val)
    {
      this.Condition = condition;
      this.Value = val;
    }

    /// <summary></summary>
    public WhereClause Condition { get; set; }
    /// <summary></summary>
    public OmExpression Value { get; set; }

  }
}

File: Qb.Js/src/CaseTerm.ts
import { WhereClause } from "./WhereClause"
import { OmExpression } from "./OmExpression"

export class CaseTerm {
  Condition: WhereClause;
  Value: OmExpression;

  constructor(condition: WhereClause, val: OmExpression) {
    this.Condition = condition;
    this.Value = val;
  }
}
